frontend/src/host_orchestrator/orchestrator/instancemanager.go

```diff
@@ -194,26 +219,34 @@ func (m *CVDToolInstanceManager) launchCVD(req apiv1.CreateCVDRequest, op apiv1.
 	result = OperationResult{Value: cvd}
 }
 
-func (m *CVDToolInstanceManager) launchCVD_(
-	req apiv1.CreateCVDRequest,
-	op apiv1.Operation) (*apiv1.CVD, error) {
-	if err := m.downloadCVDHandler.Download(); err != nil {
-		return nil, err
-	}
-	if err := createDir(m.paths.ArtifactsRootDir, false); err != nil {
+func (m *CVDToolInstanceManager) launchFromAndroidCI(
+	req apiv1.CreateCVDRequest, op apiv1.Operation) (*apiv1.CVD, error) {
+	artifactsDir, err := m.fetchCVDHandler.Fetch(req.CVD.BuildSource.AndroidCIBuild)
+	if err != nil {
 		return nil, err
 	}
-	if err := createDir(m.paths.HomesRootDir, false); err != nil {
+	instanceNumber := atomic.AddUint32(&m.instanceCounter, 1)
+	cvdName := fmt.Sprintf("cvd-%d", instanceNumber)
```

Comment:
The `cvd` tool is moving away from the "cvd-<num>" naming scheme, we shouldn't prolong its usage in here. Could we let the caller provide a name in the request instead?

Reply:
Created  b/262296730 to follow up on this.